public/app/core/components/form_dropdown/FormDropdown.tsx

```diff
@@ -0,0 +1,192 @@
+import React from 'react';
+// import ReactDOM from 'react-dom';
+import Select from 'react-select';
+import { AsyncCreatable } from 'react-select';
+import _ from 'lodash';
+import $ from 'jquery';
+import 'react-select/scss/default.scss';
+import { react2AngularDirective } from 'app/core/utils/react2angular';
+
+const NULL_VALUE_PLACEHOLDER = 'default';
+const LABEL_KEY = 'text';
+const VALUE_KEY = 'value';
+
+interface IOptionsItem {
+  value: string;
+  text: string;
+}
+
+interface IProps {
+  value: string;
+  options: IOptionsItem[];
+  cssClass: any;
+  labelMode: boolean;
+  allowCustom: boolean;
+  lookupText: boolean;
+  cache: boolean;
+  getOptions: (input: string) => any;
+  onChange: (val: any) => void;
+}
+
+export class FormDropdown extends React.Component<IProps, any> {
+  selectControl: any;
+  selectComponent: any;
+  text: string;
+  optionsCache: any;
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      value: convertToOption(this.props.value),
+      cssClasses: ""
+    };
+    this.optionsCache = {};
+
+    this.loadOptionsInternal = this.loadOptionsInternal.bind(this);
+    this.onChangeInternal = this.onChangeInternal.bind(this);
+    this.onOpen = this.onOpen.bind(this);
+    this.onClose = this.onClose.bind(this);
+    this.setSelectElement = this.setSelectElement.bind(this);
+    this.valueRenderer = this.valueRenderer.bind(this);
+  }
+
+  loadOptionsInternal(input: string) {
+    return this.props.getOptions(input)
+    .then((options) => {
+      return {
+        options: options
+      };
+    });
+  }
+
+  onChangeInternal(newValue) {
+    this.setState({value: newValue});
+    this.props.onChange(newValue);
+  }
+
+  setSelectElement(component) {
+    if (component) {
+      this.selectComponent = component.select;
+      this.selectControl = $(component.select.control);
+    } else {
+      return null;
+    }
+  }
+
+  getValueLabel() {
+    return this.selectControl.find('.Select-value-label');
+  }
+
+  onOpen() {
+    // Clear input label
+    this.getValueLabel().text("");
```

Comment:
I really like that the current value is still visible when open it so think we should remove this.